pipeline/data/importers/mono/hplt.py

```diff
@@ -0,0 +1,213 @@
+    https://data.hplt-project.org/one/monotext/cleaned/en/en_102.jsonl.zst
+    https://data.hplt-project.org/one/monotext/cleaned/en/en_103.jsonl.zst
+    https://data.hplt-project.org/one/monotext/cleaned/en/en_104.jsonl.zst
+    ...
+    https://data.hplt-project.org/one/monotext/cleaned/en/en_110.jsonl.zst
+    """
+
+    url = f"https://data.hplt-project.org/one/monotext/cleaned/{language}_map.txt"
+    logger.info(f"Downloading shard list: {url}")
+
+    with read_lines(url) as lines:
+        shard_urls = []
+        for line in lines:
+            shard_urls.append(line.strip())
+    random.Random(url).shuffle(shard_urls)
+
+    logger.info(f"Available shards for {language}:")
+    for lines in shard_urls:
+        logger.info(f" - {lines}")
+    return shard_urls
+
+
+def download_hplt(
+    language: str,
+    min_fluency_threshold: float,
+    max_lines: int,
+    max_words_in_sentence,
+    file_destination: Path,
+):
+    """
+    Downloads and filters the HPLT dataset.
+    https://hplt-project.org/datasets/v1.2
+
+    In the English data with a fluency score of 0.9, about 5% of the total sentences visited
+    are fluent enough to keep. That means to generate a monolingual dataset of 100 million
+    lines 2 billion lines need to be visited.
+
+    Parameters:
+     - language: The BCP 47 language code to filter the documents.
+     - min_fluency_threshold: The minimum score a sentence must have to be included in the final dataset.
+     - max_lines: The maximum number of lines to include in the final dataset.
+     - max_words_in_sentence: The maximum number of words allowed in each sentence.
+     - file_destination: The destination path where the final dataset will be written.
+    """
+
+    with ExitStack() as stack:
+        stats = FilteringStatistics(file_destination)
+
+        outfile = stack.enter_context(write_lines(file_destination))
+
+        shard_urls = load_shard_urls(language)
+        stats.shards.filtered = len(shard_urls)
+        document_stream = stack.enter_context(
+            read_lines(shard_urls, on_enter_location=stats.count_shards_visited)
+        )
+
+        visited_lines = 0
+
+        # Subtract 1 as the final newline is written outside of the for loop.
+        last_line = max_lines - 1
+
+        for document_json in document_stream:
+            stats.document_count.value += 1
+
+            document = HPLTDocument(**json.loads(document_json))
+
+            cumulative_word_count = 0
+            has_written_once = False
+
+            # Visit the lines in the document.
+            for score, lang_item, line in zip(document.scores, document.langs, document.lines):
+                visited_lines += 1
+
+                # Check for the fluency scores.
+                if lang_item == language and score >= min_fluency_threshold:
+                    # TODO(CJK) - Issue #424
+                    word_count = len(line.split())
```

Comment:
Heads up that this will be removed from importers and moved to the cleaning step to support CJK.